components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/JDBCPolicyDAOImpl.java

```diff
@@ -0,0 +1,1606 @@
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_ALL_PAP_POLICIES_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_ALL_PDP_POLICIES_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_LATEST_POLICY_VERSION_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PAP_POLICY_BY_VERSION_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PAP_POLICY_EDITOR_DATA_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PAP_POLICY_IDS_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PAP_POLICY_META_DATA_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PAP_POLICY_REFS_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PAP_POLICY_SET_REFS_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PAP_POLICY_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PDP_POLICY_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_POLICY_PAP_PRESENCE_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_POLICY_PDP_PRESENCE_BY_VERSION_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_POLICY_PDP_PRESENCE_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_POLICY_VERSIONS_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PUBLISHED_POLICY_VERSION_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.PUBLISH_POLICY_VERSION_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.RESTORE_ACTIVE_STATUS_AND_ORDER_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.UPDATE_ACTIVE_STATUS_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.UPDATE_ORDER_SQL;
+
+import static java.time.ZoneOffset.UTC;
+
+/**
+ * This class handles the policy operations in the JDBC data store.
+ */
+public class JDBCPolicyDAOImpl extends AbstractPolicyFinderModule implements PolicyDAO {
+
+    private static final Log LOG = LogFactory.getLog(JDBCPolicyDAOImpl.class);
+    private static final String MODULE_NAME = "JDBC Policy Finder Module";
+    private static final String ERROR_RETRIEVING_PAP_POLICY =
+            "Error while retrieving entitlement policy %s from the PAP policy store";
+    private static final String ERROR_RETRIEVING_POLICIES_FROM_POLICY_FINDER = "Policies can not be retrieved from " +
+            "the policy finder module";
+    private final int maxVersions;
+    private static final String IS_IN_PDP_1 = "IS_IN_PDP_1";
```

Comment:
The two usages of this `IS_IN_PDP_1` parameter use the value as `true` by referring to `IN_PDP`.
So do we need to parameterize this, in the first place? Can't we use the value directly in the query?

Reply:
Better to revisit all queries using constants like, `IN_PAP`, `IN_PDP`, `INACTIVE`, `DEFAULT_POLICY_ORDER`..

Without checking on the usages, everything except `DEFAULT_POLICY_ORDER` looks like the constants we can embed into the query itself.. Lets double check

Reply:
Will attend to this in a later PR.